federation/apis/federation/validation/validation.go

```diff
@@ -22,8 +22,6 @@ import (
 	"k8s.io/kubernetes/pkg/util/validation/field"
 )
 
-var ValidateClusterName = validation.NameIsDNSSubdomain
```

Comment:
Can we change Cluster name to a DNS label?  Does it need to be `foo.bar.bat` or is `foo-bar-bat` good enough?  Making it a subdomain makes it very hard to use this name in DNS ever,